src/Components/People/EditPerson.tsx

```diff
@@ -0,0 +1,199 @@
+import {
+  CardContent,
+  CardActions,
+  TextField,
+  Typography,
+  Fab,
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  SelectChangeEvent,
+} from "@mui/material";
+import DoneIcon from "@mui/icons-material/Done";
+import ClearIcon from "@mui/icons-material/Clear";
+import SaveIcon from "@mui/icons-material/Save";
+import { useEffect, useState } from "react";
+import { personEnabledDetails } from "../../Interfaces/Person";
+import GroupServices from "../../Services/Groups";
+import { toast } from "react-toastify";
+import { IGroup } from "../../Interfaces/Group";
+
+const PersonEnabledDetails = (props: personEnabledDetails) => {
+  const {
+    setAreSlotsEnabled,
+    areSlotsEnabled,
+    updatePerson,
+    createPerson,
+    name,
+    favoriteFood,
+    favoriteAnimal,
+    favoriteColor,
+    id,
+  } = props;
+  const [newName, setName] = useState(name);
+  const [newFavoriteColor, setFavoriteColor] = useState(favoriteColor);
+  const [newFavoriteAnimal, setFavoriteAnimal] = useState(favoriteAnimal);
+  const [newFavoriteFood, setFavoriteFood] = useState(favoriteFood);
```

Comment:
Instead of creating a state for each input field, you can create a single state with an object.
It'd make creating, updating and clearing your form much nicer, too.